src/ocean/text/xml/PullParser.d

```diff
@@ -543,7 +543,7 @@ version (partialwhite)
 
         final Ch[] value()
         {
-                return rawValue;
+                return rawValue.dup;
```

Comment:
Perhaps adding a note in docs about allocating behavior vs. accessing the field will be useful here.

Reply:
I've changed my mind, I am 100% certain that is the wrong approach.

That's actually a severe breaking change (much more significant than making the return value const). And it would actually break D1 as well!

Reply:
Why is this a breaking change?

Reply:
Because it means that previously working D1 code would leak memory like crazy.

We are not currently using `@nogc` but I think that we should consider all existing code which doesn't allocate memory as having an implicit `@nogc` on it. Changes which allocate memory should be considered as a breaking change.

Reply:
That's a pretty interesting point. Of course, changes that introduce new allocations are avoided, but I don't think they're strictly defined as breaking changes (from a semver perspective). But yeah, _if_ anyone is using these methods, it'd be a highly annoying change.

Reply:
It's like the 'no array stomping' change in D. The idea that it's not a breaking change, is just wrong. I think given the importance of memory efficiency to our libraries, we need to have this extra nuance compared to normal semver. But note that if we were using `@nogc` it would actually be normal semver, I think.

In practice, it is important: if this change is made, then you need to change your code, quite extensively. Although existing code will compile and passes the unit tests, it will fail burn-in tests.